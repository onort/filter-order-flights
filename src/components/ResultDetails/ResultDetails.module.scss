@import "../../styles/main.scss";

.row {
  display: grid;
  background-color: $element-bg;
  grid-column-gap: map-get($spacers, normal);
  grid-template-columns: 2fr 1fr !important;
}

.legs,
.prices {
  background-color: $white;
  border-radius: px-to-rem(2px);
}

.legs {
  padding: 0 !important;
}

.legRow {
  display: grid;
  padding: 0 !important;
  border-bottom: none !important;
  grid-template-columns: 1fr px-to-rem(170px) 1fr 2fr !important;
}

.priceOption {
  display: flex;
  margin-bottom: map-get($spacers, normal);

  span {
    display: block;
    color: $grey2;
    font-family: $font-family-primary;
    font-size: map-get($font-sizes, x-small);
    font-weight: 400;

    &.price {
      flex: 4;
      font-weight: 600;
    }

    &.agent {
      flex: 7;
    }
  }
}

.durationContainer {
  text-align: center;
}

.duration {
  color: $grey2;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
  font-weight: 400;
}

.pointedLine {
  position: relative;
  padding-bottom: 15px;
  border-bottom: 1px solid $grey2;
}

.pointedLine::after,
.pointedLine::before {
  position: absolute;
  bottom: -3px;
  left: 0;
  width: 5px;
  height: 5px;
  background: $grey2;
  border-radius: 50%;
  content: "";
}

.pointedLine::after {
  right: 0;
  left: auto;
}
