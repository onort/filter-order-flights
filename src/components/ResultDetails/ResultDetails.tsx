import React from "react"
import dayjs from "dayjs"

import styles from "./ResultDetails.module.scss"
import { Leg, PricingOption } from "../../types"
import { Cell, DataCell, Row } from "../"
import { minsToString } from "../../utils"

interface Props {
  legs: Leg[]
  pricingOptions: PricingOption[]
}

const ResultDetails: React.SFC<Props> = props => {
  return (
    <Row className={styles.row}>
      <Cell className={styles.legs}>
        {props.legs.map(leg => (
          <Row key={leg.flightCode} className={styles.legRow}>
            <DataCell
              sub={`${leg.departure.airportCode} ${leg.departure.airportName}`}
              val={dayjs(leg.departure.time).format("HH:mm")}
            />
            <Cell className={styles.durationContainer}>
              <span className={styles.duration}>
                {minsToString(leg.duration)}
              </span>
              <div className={styles.pointedLine} />
            </Cell>
            <DataCell
              sub={`${leg.arrival.airportCode} ${leg.arrival.airportName}`}
              val={dayjs(leg.arrival.time).format("HH:mm")}
            />
            <DataCell sub={leg.flightCode} val={leg.carrier} />
          </Row>
        ))}
      </Cell>
      <Cell className={styles.prices}>
        {props.pricingOptions.map(option => (
          <div key={option.agent} className={styles.priceOption}>
            <span className={styles.price}>{option.price} TL</span>
            <span className={styles.agent}>&#64;{option.agent}</span>
          </div>
        ))}
      </Cell>
    </Row>
  )
}

export default ResultDetails
