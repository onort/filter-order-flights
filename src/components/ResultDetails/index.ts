import ResultDetails from "./ResultDetails"

export default ResultDetails
