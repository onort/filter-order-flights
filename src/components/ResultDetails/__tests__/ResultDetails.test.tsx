import React from "react"
import { mount, shallow, ShallowWrapper } from "enzyme"

import ResultDetails from "../"

const legs = [
  {
    arrival: {
      airportCode: "ADB",
      airportName: "Izmir",
      time: "2017-04-14T09:10:00"
    },
    carrier: "Pegasus Airlines",
    departure: {
      airportCode: "ADA",
      airportName: "Adana",
      time: "2017-04-14T07:40:00"
    },
    duration: 90,
    flightCode: "PC 3011"
  }
]

const pricingOptions = [
  {
    agent: "Pegasus Airlines",
    price: 66.99
  },
  {
    agent: "AeroBilet",
    price: 70.34
  }
]

describe("<ResultDetails />", () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(
      <ResultDetails legs={legs} pricingOptions={pricingOptions} />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct amount of legRows", () => {
    expect(wrapper.find(".legRow")).toHaveLength(1)
  })

  it("should have correct amount of priceOptions", () => {
    expect(wrapper.find(".priceOption")).toHaveLength(2)
  })
})
