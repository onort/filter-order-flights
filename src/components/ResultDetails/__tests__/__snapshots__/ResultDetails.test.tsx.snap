// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultDetails /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultDetails
    legs={
      Array [
        Object {
          "arrival": Object {
            "airportCode": "ADB",
            "airportName": "Izmir",
            "time": "2017-04-14T09:10:00",
          },
          "carrier": "Pegasus Airlines",
          "departure": Object {
            "airportCode": "ADA",
            "airportName": "Adana",
            "time": "2017-04-14T07:40:00",
          },
          "duration": 90,
          "flightCode": "PC 3011",
        },
      ]
    }
    pricingOptions={
      Array [
        Object {
          "agent": "Pegasus Airlines",
          "price": 66.99,
        },
        Object {
          "agent": "AeroBilet",
          "price": 70.34,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Cell
          className="legs"
        >
          <Row
            className="legRow"
          >
            <DataCell
              className="origin"
              detailCell={true}
              sub="ADA Adana"
              val="07:40"
            />
            <Cell
              className="durationContainer"
            >
              <span
                className="duration"
              >
                1 saat 30 dakika
              </span>
              <div
                className="pointedLine"
              />
            </Cell>
            <DataCell
              className="destination"
              detailCell={true}
              sub="Izmir ADB "
              val="09:10"
            />
            <DataCell
              className="flightInfo"
              detailCell={true}
              sub="PC 3011 Ekonomi"
              val="Pegasus Airlines"
            />
          </Row>
        </Cell>,
        <Cell
          className="prices"
        >
          <div
            className="priceOption"
          >
            <span
              className="price"
            >
              66.99
               TL
            </span>
            <span
              className="agent"
            >
              @
              Pegasus Airlines
            </span>
          </div>
          <div
            className="priceOption"
          >
            <span
              className="price"
            >
              70.34
               TL
            </span>
            <span
              className="agent"
            >
              @
              AeroBilet
            </span>
          </div>
        </Cell>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Row
              className="legRow"
            >
              <DataCell
                className="origin"
                detailCell={true}
                sub="ADA Adana"
                val="07:40"
              />
              <Cell
                className="durationContainer"
              >
                <span
                  className="duration"
                >
                  1 saat 30 dakika
                </span>
                <div
                  className="pointedLine"
                />
              </Cell>
              <DataCell
                className="destination"
                detailCell={true}
                sub="Izmir ADB "
                val="09:10"
              />
              <DataCell
                className="flightInfo"
                detailCell={true}
                sub="PC 3011 Ekonomi"
                val="Pegasus Airlines"
              />
            </Row>,
          ],
          "className": "legs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "PC 3011",
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <DataCell
                  className="origin"
                  detailCell={true}
                  sub="ADA Adana"
                  val="07:40"
                />,
                <Cell
                  className="durationContainer"
                >
                  <span
                    className="duration"
                  >
                    1 saat 30 dakika
                  </span>
                  <div
                    className="pointedLine"
                  />
                </Cell>,
                <DataCell
                  className="destination"
                  detailCell={true}
                  sub="Izmir ADB "
                  val="09:10"
                />,
                <DataCell
                  className="flightInfo"
                  detailCell={true}
                  sub="PC 3011 Ekonomi"
                  val="Pegasus Airlines"
                />,
              ],
              "className": "legRow",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "origin",
                  "detailCell": true,
                  "sub": "ADA Adana",
                  "val": "07:40",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <span
                      className="duration"
                    >
                      1 saat 30 dakika
                    </span>,
                    <div
                      className="pointedLine"
                    />,
                  ],
                  "className": "durationContainer",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1 saat 30 dakika",
                      "className": "duration",
                    },
                    "ref": null,
                    "rendered": "1 saat 30 dakika",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "pointedLine",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "destination",
                  "detailCell": true,
                  "sub": "Izmir ADB ",
                  "val": "09:10",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "flightInfo",
                  "detailCell": true,
                  "sub": "PC 3011 Ekonomi",
                  "val": "Pegasus Airlines",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="priceOption"
            >
              <span
                className="price"
              >
                66.99
                 TL
              </span>
              <span
                className="agent"
              >
                @
                Pegasus Airlines
              </span>
            </div>,
            <div
              className="priceOption"
            >
              <span
                className="price"
              >
                70.34
                 TL
              </span>
              <span
                className="agent"
              >
                @
                AeroBilet
              </span>
            </div>,
          ],
          "className": "prices",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Pegasus Airlines",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="price"
                >
                  66.99
                   TL
                </span>,
                <span
                  className="agent"
                >
                  @
                  Pegasus Airlines
                </span>,
              ],
              "className": "priceOption",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    66.99,
                    " TL",
                  ],
                  "className": "price",
                },
                "ref": null,
                "rendered": Array [
                  66.99,
                  " TL",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "@",
                    "Pegasus Airlines",
                  ],
                  "className": "agent",
                },
                "ref": null,
                "rendered": Array [
                  "@",
                  "Pegasus Airlines",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "AeroBilet",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="price"
                >
                  70.34
                   TL
                </span>,
                <span
                  className="agent"
                >
                  @
                  AeroBilet
                </span>,
              ],
              "className": "priceOption",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    70.34,
                    " TL",
                  ],
                  "className": "price",
                },
                "ref": null,
                "rendered": Array [
                  70.34,
                  " TL",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "@",
                    "AeroBilet",
                  ],
                  "className": "agent",
                },
                "ref": null,
                "rendered": Array [
                  "@",
                  "AeroBilet",
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Cell
            className="legs"
          >
            <Row
              className="legRow"
            >
              <DataCell
                className="origin"
                detailCell={true}
                sub="ADA Adana"
                val="07:40"
              />
              <Cell
                className="durationContainer"
              >
                <span
                  className="duration"
                >
                  1 saat 30 dakika
                </span>
                <div
                  className="pointedLine"
                />
              </Cell>
              <DataCell
                className="destination"
                detailCell={true}
                sub="Izmir ADB "
                val="09:10"
              />
              <DataCell
                className="flightInfo"
                detailCell={true}
                sub="PC 3011 Ekonomi"
                val="Pegasus Airlines"
              />
            </Row>
          </Cell>,
          <Cell
            className="prices"
          >
            <div
              className="priceOption"
            >
              <span
                className="price"
              >
                66.99
                 TL
              </span>
              <span
                className="agent"
              >
                @
                Pegasus Airlines
              </span>
            </div>
            <div
              className="priceOption"
            >
              <span
                className="price"
              >
                70.34
                 TL
              </span>
              <span
                className="agent"
              >
                @
                AeroBilet
              </span>
            </div>
          </Cell>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                className="legRow"
              >
                <DataCell
                  className="origin"
                  detailCell={true}
                  sub="ADA Adana"
                  val="07:40"
                />
                <Cell
                  className="durationContainer"
                >
                  <span
                    className="duration"
                  >
                    1 saat 30 dakika
                  </span>
                  <div
                    className="pointedLine"
                  />
                </Cell>
                <DataCell
                  className="destination"
                  detailCell={true}
                  sub="Izmir ADB "
                  val="09:10"
                />
                <DataCell
                  className="flightInfo"
                  detailCell={true}
                  sub="PC 3011 Ekonomi"
                  val="Pegasus Airlines"
                />
              </Row>,
            ],
            "className": "legs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "PC 3011",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <DataCell
                    className="origin"
                    detailCell={true}
                    sub="ADA Adana"
                    val="07:40"
                  />,
                  <Cell
                    className="durationContainer"
                  >
                    <span
                      className="duration"
                    >
                      1 saat 30 dakika
                    </span>
                    <div
                      className="pointedLine"
                    />
                  </Cell>,
                  <DataCell
                    className="destination"
                    detailCell={true}
                    sub="Izmir ADB "
                    val="09:10"
                  />,
                  <DataCell
                    className="flightInfo"
                    detailCell={true}
                    sub="PC 3011 Ekonomi"
                    val="Pegasus Airlines"
                  />,
                ],
                "className": "legRow",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "origin",
                    "detailCell": true,
                    "sub": "ADA Adana",
                    "val": "07:40",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <span
                        className="duration"
                      >
                        1 saat 30 dakika
                      </span>,
                      <div
                        className="pointedLine"
                      />,
                    ],
                    "className": "durationContainer",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "1 saat 30 dakika",
                        "className": "duration",
                      },
                      "ref": null,
                      "rendered": "1 saat 30 dakika",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "pointedLine",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "destination",
                    "detailCell": true,
                    "sub": "Izmir ADB ",
                    "val": "09:10",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "flightInfo",
                    "detailCell": true,
                    "sub": "PC 3011 Ekonomi",
                    "val": "Pegasus Airlines",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                className="priceOption"
              >
                <span
                  className="price"
                >
                  66.99
                   TL
                </span>
                <span
                  className="agent"
                >
                  @
                  Pegasus Airlines
                </span>
              </div>,
              <div
                className="priceOption"
              >
                <span
                  className="price"
                >
                  70.34
                   TL
                </span>
                <span
                  className="agent"
                >
                  @
                  AeroBilet
                </span>
              </div>,
            ],
            "className": "prices",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Pegasus Airlines",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="price"
                  >
                    66.99
                     TL
                  </span>,
                  <span
                    className="agent"
                  >
                    @
                    Pegasus Airlines
                  </span>,
                ],
                "className": "priceOption",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      66.99,
                      " TL",
                    ],
                    "className": "price",
                  },
                  "ref": null,
                  "rendered": Array [
                    66.99,
                    " TL",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "@",
                      "Pegasus Airlines",
                    ],
                    "className": "agent",
                  },
                  "ref": null,
                  "rendered": Array [
                    "@",
                    "Pegasus Airlines",
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "AeroBilet",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="price"
                  >
                    70.34
                     TL
                  </span>,
                  <span
                    className="agent"
                  >
                    @
                    AeroBilet
                  </span>,
                ],
                "className": "priceOption",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      70.34,
                      " TL",
                    ],
                    "className": "price",
                  },
                  "ref": null,
                  "rendered": Array [
                    70.34,
                    " TL",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "@",
                      "AeroBilet",
                    ],
                    "className": "agent",
                  },
                  "ref": null,
                  "rendered": Array [
                    "@",
                    "AeroBilet",
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ResultDetails /> should match snapshot 2`] = `"<div class=\\"row row\\"><div class=\\"cell legs\\"><div class=\\"row legRow\\"><div class=\\"cell dataCell detailCell origin\\"><span class=\\"val\\">07:40</span><span class=\\"sub\\">ADA Adana</span></div><div class=\\"cell durationContainer\\"><span class=\\"duration\\">1 saat 30 dakika</span><div class=\\"pointedLine\\"></div></div><div class=\\"cell dataCell detailCell destination\\"><span class=\\"val\\">09:10</span><span class=\\"sub\\">Izmir ADB </span></div><div class=\\"cell dataCell detailCell flightInfo\\"><span class=\\"val\\">Pegasus Airlines</span><span class=\\"sub\\">PC 3011 Ekonomi</span></div></div></div><div class=\\"cell prices\\"><div class=\\"priceOption\\"><span class=\\"price\\">66.99 TL</span><span class=\\"agent\\">@Pegasus Airlines</span></div><div class=\\"priceOption\\"><span class=\\"price\\">70.34 TL</span><span class=\\"agent\\">@AeroBilet</span></div></div></div>"`;
